### PR TITLE
Es6 projects common modules discussion upvote

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

ES6

### CODE REVIEW

**Review by gu-stav — APPROVED** [review]
Fantastique.

File: static/src/javascripts/projects/common/modules/discussion/upvote.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    displayName: string,
+    webUrl: string,
+    apiUrl: string,
+    avatar: string,
+    secureAvatarUrl: string,
+    badge: Array<string>,
+    details: {
+        gender: string,
+    },
+};
```
Comment (gustavpursche) [inline static/src/javascripts/projects/common/modules/discussion/upvote.js]: Mini nit: could you move this to the top of the file?

File: static/src/javascripts-legacy/projects/common/modules/discussion/loader.js [hunk 2]
```diff
@@ -249,7 +249,7 @@ Loader.prototype.initToolbar = function() {
 
 Loader.prototype.initRecommend = function() {
     this.on('click', '.js-recommend-comment', function(e) {
-        upvote.handle(e.currentTarget, this.elem, this.user, DiscussionApi, config.switches.discussionAllowAnonymousRecommendsSwitch);
```
Comment (SiAdcock) [inline static/src/javascripts-legacy/projects/common/modules/discussion/loader.js]: Passing these last two params is unnecessary as they are singletons and can be directly imported into the `upvote` module

File: static/src/javascripts/projects/common/modules/discussion/upvote.js (hunk 1 above)
Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/discussion/upvote.js]: These props are the minimum we get back from hitting the `profile/:id` endpoint on Discussion API. We can expand this type later if necessary, but we don't need much detail in this module.

**Pull request merged**